File: crates/algosketch-cli/fixtures/binary_search.java
class Algorithms {
    int binary_search(int[] items, int target) {
        int low = 0;
        int high = items.length - 1;
        while (low <= high) {
            int mid = (low + high) / 2;
            if (items[mid] == target) {
                return mid;
            } else if (items[mid] < target) {
                low = mid + 1;
            } else {
                high = mid - 1;
            }
        }
        return -1;
    }
}

File: crates/algosketch-cli/fixtures/binary_search.go
package main

func binary_search(items []int, target int) int {
	low := 0
	high := len(items) - 1
	for low <= high {
		mid := (low + high) / 2
		if items[mid] == target {
			return mid
		} else if items[mid] < target {
			low = mid + 1
		} else {
			high = mid - 1
		}
	}
	return -1
}
